src/Reflection/SignatureMap/NativeFunctionReflectionProvider.php

```diff
@@ -49,16 +56,24 @@ public function findFunctionReflection(string $functionName): ?NativeFunctionRef
 			return null;
 		}
 
+		$phpDoc = $this->stubPhpDocProvider->findFunctionPhpDoc($lowerCasedFunctionName);
+
 		$variants = [];
 		$i = 0;
 		while ($this->signatureMapProvider->hasFunctionSignature($lowerCasedFunctionName, $i)) {
 			$functionSignature = $this->signatureMapProvider->getFunctionSignature($lowerCasedFunctionName, null, $i);
-			$returnType = $functionSignature->getReturnType();
 			$variants[] = new FunctionVariant(
 				TemplateTypeMap::createEmpty(),
 				null,
-				array_map(static function (ParameterSignature $parameterSignature) use ($lowerCasedFunctionName): NativeParameterReflection {
+				array_map(static function (ParameterSignature $parameterSignature) use ($lowerCasedFunctionName, $phpDoc): NativeParameterReflection {
 					$type = $parameterSignature->getType();
+					$phpDocType = null;
+					if ($phpDoc !== null) {
+						$phpDocParam = $phpDoc->getParamTags()[$parameterSignature->getName()] ?? null;
```

Comment:
Unfortunately this needs more complicated logic - the parameter names change across PHP versions so the stub will definitely be wrong and the PHPDoc unused. We can wait and see for the tests - I suspect that different tests will fail on different PHP versions.

Reply:
A similar logic is already applied when inheriting PHPDocs implicitly - parameter names can change in child classes and PHPDocs are mapped based on that.

`StubPhpDocProvider::findFunctionPhpDoc()` would need a similar logic like `findMethodPhpDoc` already has - the `$positionalParameterNames` parameter and calling `changeParameterNamesByMapping`.